pkg/assembler/backends/arangodb/certifyBad.go

```diff
@@ -956,54 +917,29 @@ func getArtifactCertifyBad(ctx context.Context, cursor driver.Cursor) ([]*model.
 
 	var certifyBadList []*model.CertifyBad
 	for _, createdValue := range createdValues {
+		var pkg *model.Package = nil
+		var src *model.Source = nil
+		if createdValue.PkgVersion != nil {
+			pkg = generateModelPackage(createdValue.PkgVersion.TypeID, createdValue.PkgVersion.PkgType, createdValue.PkgVersion.NamespaceID, createdValue.PkgVersion.Namespace, createdValue.PkgVersion.NameID,
+				createdValue.PkgVersion.Name, createdValue.PkgVersion.VersionID, createdValue.PkgVersion.Version, createdValue.PkgVersion.Subpath, createdValue.PkgVersion.QualifierList)
+		} else if createdValue.SrcName != nil {
+			src = generateModelSource(createdValue.SrcName.TypeID, createdValue.SrcName.SrcType, createdValue.SrcName.NamespaceID, createdValue.SrcName.Namespace,
+				createdValue.SrcName.NameID, createdValue.SrcName.Name, createdValue.SrcName.Commit, createdValue.SrcName.Tag)
+		}
+
 		certifyBad := &model.CertifyBad{
 			ID:            createdValue.CertifyBadID,
-			Subject:       createdValue.Artifact,
 			Justification: createdValue.Justification,
 			Origin:        createdValue.Collector,
 			Collector:     createdValue.Origin,
 		}
-		certifyBadList = append(certifyBadList, certifyBad)
-	}
-	return certifyBadList, nil
-}
-
-func getSourceCertifyBad(ctx context.Context, cursor driver.Cursor) ([]*model.CertifyBad, error) {
-	type collectedData struct {
-		SrcName       *dbSrcName `json:"srcName"`
-		CertifyBadID  string     `json:"certifyBad_id"`
-		Justification string     `json:"justification"`
-		Collector     string     `json:"collector"`
-		Origin        string     `json:"origin"`
-	}
 
-	var createdValues []collectedData
-	for {
-		var doc collectedData
-		_, err := cursor.ReadDocument(ctx, &doc)
-		if err != nil {
-			if driver.IsNoMoreDocuments(err) {
-				break
-			} else {
-				return nil, fmt.Errorf("failed to source certifyBad from cursor: %w", err)
-			}
+		if pkg != nil {
+			certifyBad.Subject = pkg
+		} else if src != nil {
+			certifyBad.Subject = src
 		} else {
-			createdValues = append(createdValues, doc)
-		}
-	}
-
-	var certifyBadList []*model.CertifyBad
-	for _, createdValue := range createdValues {
-
-		src := generateModelSource(createdValue.SrcName.TypeID, createdValue.SrcName.SrcType, createdValue.SrcName.NamespaceID, createdValue.SrcName.Namespace,
-			createdValue.SrcName.NameID, createdValue.SrcName.Name, createdValue.SrcName.Commit, createdValue.SrcName.Tag)
-
-		certifyBad := &model.CertifyBad{
-			ID:            createdValue.CertifyBadID,
-			Subject:       src,
-			Justification: createdValue.Justification,
-			Origin:        createdValue.Collector,
-			Collector:     createdValue.Origin,
+			certifyBad.Subject = createdValue.Artifact
```

Comment:
nit: wonder if it would be helpful to check if artifact is nil as well and if so throw an error